docs_src/microservices/application/V2Migration.md

```diff
@@ -0,0 +1,298 @@
+    ```toml
+      [Clients]
+        # Used for version check on start-up
+        # Also used for DeviceService, DeviceProfile and Device clients
+        [Clients.core-metadata]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59881
+        
+    ```
+    
+        # Used for Event client which is used by PushToCoreData function
+        [Clients.core-data]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59880
+        
+        # Used for Command client
+        [Clients.core-command]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59882
+        
+        # Used for Notification and Subscription clients
+        [Clients.support-notifications]
+        Protocol = 'http'
+        Host = 'localhost'
+        Port = 59860
+    ```
+
+#### Trigger
+
+The `Trigger` section (previously named `Binding`) has been restructured with `EdgexMessageBus` (previously named `MessageBus`) and `ExternalMqtt` (previously named `MqttBroker` ) moved under it. The `SubscribeTopics` (previously named `SubscribeTopic`) has been moved under the `EdgexMessageBus.SubscribeHost` and `ExternalMqtt` sections. The `PublishTopic` has been moved under the `EdgexMessageBus.PublishHost` and `ExternalMqtt` sections.
+
+##### EdgeX MessageBus
+
+If your Application Service is using the EdgeX MessageBus trigger you can then simply copy the complete `Trigger` configuration from the example below and tweak it as needed. 
+
+!!! example "Example - EdgeX MessageBus trigger configuration"
+
+    ```toml
+    [Trigger]
+    Type="edgex-messagebus"
+      [Trigger.EdgexMessageBus]
+      Type = "redis"
+        [Trigger.EdgexMessageBus.SubscribeHost]
+        Host = "localhost"
+        Port = 6379
+        Protocol = "redis"
+        SubscribeTopics="edgex/events/#"
+        [Trigger.EdgexMessageBus.PublishHost]
+        Host = "localhost"
+        Port = 6379
+        Protocol = "redis"
+        PublishTopic="example"
+        [Trigger.EdgexMessageBus.Optional]
+        AuthMode = 'usernamepassword'  # required for redis messagebus (secure or insecure).
+        SecretName = 'redisdb'
+    ```
+
+From the above example you can see the improved structure and the following changes:
+
+- Default `EdgexMessageBus` type has changed from `ZeroMQ` to `Redis`.
+- Type value for `Redis` has changed from `redistreams` to `redis`. This is because the implementation no longer uses Redis Streams. It now uses Redis Pub/Sub.
+- `SubscribeTopics` is now plural since it now accepts a comma separated list of topics. The default value uses a multi-level topic with a wild card. This is because Core Data and Device Services now publish to a multi-level topics which have`edgex/events` as their base. This allows Application Services to filter by topic rather then receive the data and then filter it out via a pipeline filter function. See the [Filter By Topics](../Triggers/#filter-by-topics) section for more details.
```

Comment:
We attempted standardized on `#` as the wild card for the EdgeX MessageBus, The `#` wildcard is being converted to `*` in the Redis implementation of the EdgeX MessageBus. Since we assumed `#` is wildcard for MQTT at all levels we do not do any conversions in the MQTT implementation. 

So, for MQTT, the subscribe topics follow the standard MQTT  multi (`#`) and single (`+`) level wild cards. I will add a note to this section to make this explicit.

Reply:
> * What happens if I add my own [glob-style wildcards](https://redis.io/commands/psubscribe) when using redis or other brokers?

This would break for Redis if the `*` is used. Others should still work.

Reply:
I will create a separate PR with the update to the `Filter By Topics` section

Reply:
Makes sense. Thanks for the explanation.

> Since we assumed `#` is wildcard for MQTT at all levels we do not do any conversions in the MQTT implementation.
> 
> So, for MQTT, the subscribe topics follow the standard MQTT multi (`#`) and single (`+`) level wild cards.

So are you saying that when using MQTT style, it does no conversion and it is up to the user to use the `+` vs the `#`? I ask mainly because one main gotcha of MQTT wildcards are that the `#` is only allowed once and only at the end of the topic, so things like `edgex/events/#/Uptime` are invalid.

Example from [Trigger Docs](https://docs.edgexfoundry.org/2.0/microservices/application/Triggers/)
> **Filter for Events only from a single actual device**
> SubscribeTopics="edgex/events/#/#/trendnet01/#"

In MQTT this would be invalid and have to be either. `"edgex/events/+/+/trendnet01/#"` or `"edgex/events/+/+/trendnet01/+"`

Reply:
> So are you saying that when using MQTT style, it does no conversion and it is up to the user to use the `+` vs the `#`? I ask mainly because one main gotcha of MQTT wildcards are that the `#` is only allowed once and only at the end of the topic, so things like `edgex/events/#/Uptime` are invalid.

Yep. I ran a test this morning and verified that. Clarification add in this PR: https://github.com/edgexfoundry/edgex-docs/pull/462